sklearn/cross_validation.py

```diff
@@ -332,6 +332,119 @@ def __len__(self):
         return self.n_folds
 
 
+def disjoint_label_folds(y, n_folds=3):
+    """Creates folds where a same label is not in two different folds.
+    
+    Parameters
+    ----------
+    y: numpy array, shape (n_samples,)
+        Contains an id for each sample.
+        The folds are built so that the same id doesn't appear in two different folds.
```

Comment:
@larsmans @ogrisel changed to mergesort, thanks.
@larsmans currently the implementation allows the user to have an array of strings for y (for instance the name or id of the subjects in an experiment) as I didn't see any reason to not allow that. Do you want to enforce a np.intp dtype?